lib/src/configuration.dart

```diff
@@ -70,65 +70,197 @@ class Configuration {
   /// Specifies whether a [Realm] should be opened as read-only.
+  ///
   /// This allows opening it from locked locations such as resources,
   /// bundled with an application.
   ///
   /// The realm file must already exists at [path]
   final bool isReadOnly;
 
-  /// Specifies whether a [Realm] should be opened in-memory.
-  ///
-  /// This still requires a [path] (can be the default path) to identify the [Realm] so other processes can open the same [Realm].
-  /// The file will also be used as swap space if the [Realm] becomes bigger than what fits in memory,
-  /// but it is not persistent and will be removed when the last instance is closed.
-  /// When all in-memory instance of [Realm] is closed all data in that [Realm] is deleted.
-  final bool isInMemory;
-
-  /// Specifies the FIFO special files fallback location.
-  /// Opening a [Realm] creates a number of FIFO special files in order to
-  /// coordinate access to the [Realm] across threads and processes. If the [Realm] file is stored in a location
-  /// that does not allow the creation of FIFO special files (e.g. FAT32 filesystems), then the [Realm] cannot be opened.
-  /// In that case [Realm] needs a different location to store these files and this property defines that location.
-  /// The FIFO special files are very lightweight and the main [Realm] file will still be stored in the location defined
-  /// by the [path] you  property. This property is ignored if the directory defined by [path] allow FIFO special files.
-  final String? fifoFilesFallbackPath;
-
   /// Specifies if a [Realm] file format should be automatically upgraded
   /// if it was created with an older version of the [Realm] library.
   /// An exception will be thrown if a file format upgrade is required.
   final bool disableFormatUpgrade;
 
-  /// A function that will be executed only when the Realm is first created.
-  ///
-  /// The Realm instance passed in the callback already has a write transaction opened, so you can
-  /// add some initial data that your app needs. The function will not execute for existing
-  /// Realms, even if all objects in the Realm are deleted.
-  final Function(Realm realm)? initialDataCallback;
-
-  /// The function called when opening a Realm for the first time
-  /// during the life of a process to determine if it should be compacted
-  /// before being returned to the user.
-  ///
-  /// [totalSize] - The total file size (data + free space)
-  /// [usedSize] - The total bytes used by data in the file.
-  /// It returns true to indicate that an attempt to compact the file should be made.
-  /// The compaction will be skipped if another process is currently accessing the realm file.
-  final bool Function(int totalSize, int usedSize)? shouldCompactCallback;
+  /// The function will be called when opening a [Realm] for the first time
+  /// during the life of a process.
+  final ShouldCompactCallback? shouldCompactCallback;
+
+  final InitialDataCallback? initialDataCallback;
```

Comment:
Missing documentation. It will be good to refer also to [InitialDataCallback] doc description.